src/accordion/accordion.directive.ts

```diff
@@ -0,0 +1,305 @@
+		{
+			directive: NgbCollapse,
+		},
+	],
+})
+export class NgbAccordionCollapse {
+	constructor(
+		@Inject(forwardRef(() => NgbAccordionItem)) public item: NgbAccordionItem,
+		public ngbCollapse: NgbCollapse,
+	) {}
+}
+
+@Directive({
+	selector: '[ngbAccordionToggle]',
+	standalone: true,
+	host: {
+		'[id]': 'item.toggleId',
+		'[disabled]': 'item.disabled',
+		'[class.accordion-button]': 'true',
+		'[class.collapsed]': 'item.collapsed',
+		'[attr.aria-controls]': 'item.collapseId',
+		'[attr.aria-expanded]': '!item.collapsed',
+		'[type]': '"button"',
+		'(click)': 'accordion.toggle(item.id)',
+	},
+})
+export class NgbAccordionToggle {
+	constructor(
+		@Inject(forwardRef(() => NgbAccordionItem)) public item: NgbAccordionItem,
+		@Inject(forwardRef(() => NgbAccordionDirective)) public accordion: NgbAccordionDirective,
+	) {}
+}
+
+@Directive({
+	selector: '[ngbAccordionHeader]',
+	standalone: true,
+	host: {
+		role: 'heading',
+		'[class.accordion-header]': 'true',
+	},
+})
+export class NgbAccordionHeader {
+	constructor() {}
+}
+
+@Directive({
+	selector: '[ngbAccordionItem]',
+	standalone: true,
+	host: {
+		'[class.accordion-item]': 'true',
+		'[id]': 'id',
+	},
+})
+export class NgbAccordionItem implements AfterContentInit {
+	constructor(
+		@Inject(forwardRef(() => NgbAccordionDirective)) private accordion: NgbAccordionDirective,
+		private cd: ChangeDetectorRef,
+	) {}
+
+	private subscriptions: Subscription[] = [];
+	private _collapsed = true;
+	private _id = `${nextId++}`;
+	private _disabled = false;
+
+	animatingBodyCollapse = false;
+
+	@ContentChild(NgbAccordionCollapse, { static: true }) private collapse: NgbAccordionCollapse;
+
+	@Input('ngbAccordionItem') set id(inputId: string) {
+		if (inputId) {
+			this._id = inputId;
+		}
+	}
+	@Input() destroyOnHide = this.accordion.destroyOnHide;
+	@Input() set disabled(isDisabled: boolean) {
+		if (isDisabled != this.disabled) {
+			this._disabled = isDisabled;
+			if (this._disabled) {
+				this._collapsed = true;
+				this.collapse.ngbCollapse.collapsed = true;
+			}
+		}
+	}
+	@Input() set collapsed(collapsed: boolean) {
+		if (this.disabled) {
+			if (!this.collapsed) {
+				this._collapsed = true;
+			}
+		} else if (this.collapsed !== collapsed) {
+			this._collapsed = collapsed;
+			// we need force CD to get template into DOM before starting animation to calculate its height correctly
+			if (!this._collapsed) {
+				this.animatingBodyCollapse = true;
+				this.cd.detectChanges();
+			}
+			// we also need to make sure 'animation' flag is up-to- date
+			this.collapse.ngbCollapse.animation = this.accordion.animation;
+			this.collapse.ngbCollapse.collapsed = collapsed;
+		}
+	}
+
+	@Output() shown = new EventEmitter<void>();
+	@Output() hidden = new EventEmitter<void>();
+
+	get collapsed(): boolean {
+		return this._collapsed;
+	}
+	get disabled() {
+		return this._disabled;
+	}
+	public get id() {
+		return `ngb-accordion-item-${this._id}`;
+	}
+
+	get toggleId() {
```

Comment:
Maybe just `{this.id}-toggle` (note the `id`, not `_id`) ?

It will give:

```
ngb-accordion-item-1
ngb-accordion-item-1-toggle
ngb-accordion-item-1-collapse
```

And I think it is better, because toggle and collapse are nested in the item